test/engine/SpatialJoinAlgorithmsTest.cpp

```diff
@@ -780,8 +781,277 @@ TEST_P(SpatialJoinParamTest, computeResultSmallDatasetDifferentSizeChildren) {
       expectedMaxDist10000000_rows_diff, columnNames, false);
 }
 
-INSTANTIATE_TEST_SUITE_P(SpatialJoin, SpatialJoinParamTest, ::testing::Bool());
+INSTANTIATE_TEST_SUITE_P(
+    SpatialJoin, SpatialJoinParamTest,
+    ::testing::Values(SpatialJoin::Algorithm::Baseline,
+                      SpatialJoin::Algorithm::S2Geometry,
+                      SpatialJoin::Algorithm::BoundingBox));
 
 }  // end of Namespace computeResultTest
 
+namespace boundingBox {
+
+namespace bg = boost::geometry;
+namespace bgi = boost::geometry::index;
+
+typedef bg::model::point<double, 2, bg::cs::cartesian> point;
+typedef bg::model::box<point> box;
+typedef std::pair<point, size_t> value;
+
+inline void testBoundingBox(const long long& maxDistInMeters,
+                            const point& startPoint) {
+  auto checkOutside = [&](const point& point1, const point& startPoint,
+                          const std::vector<box>& bbox,
+                          SpatialJoinAlgorithms* spatialJoinAlg) {
+    // check if the point is contained in any bounding box
+    bool within = spatialJoinAlg->OnlyForTestingWrapperContainedInBoundingBoxes(
+        bbox, point1);
+    if (!within) {
+      GeoPoint geo1{point1.get<1>(), point1.get<0>()};
+      GeoPoint geo2{startPoint.get<1>(), startPoint.get<0>()};
+      double dist = ad_utility::detail::wktDistImpl(geo1, geo2) * 1000;
+      ASSERT_GT(static_cast<long long>(dist), maxDistInMeters);
```

Comment:
In some future PR you could make the maxDistance inside your SpatialJoinclass a `double`, this is more natural and also avoids many of those casts.

Reply:
yes that's true